media/server/gstplayer/source/GstGenericPlayer.cpp

```diff
@@ -478,33 +478,26 @@ bool GstGenericPlayer::setImmediateOutput(const MediaSourceType &mediaSourceType
     if (!m_workerThread)
         return false;
 
-    m_workerThread->enqueueTask(m_taskFactory->createSetImmediateOutput(*this, mediaSourceType, immediateOutputParam));
+    m_workerThread->enqueueTask(
+        m_taskFactory->createSetImmediateOutput(m_context, *this, mediaSourceType, immediateOutputParam));
     return true;
 }
 
 bool GstGenericPlayer::getImmediateOutput(const MediaSourceType &mediaSourceType, bool &immediateOutputRef)
 {
     bool returnValue{false};
-    GstElement *sink = getSink(mediaSourceType);
-    if (sink)
+    ActualSink sink(*this, mediaSourceType, m_gstWrapper);
+    if (sink.get())
```

Comment:
if we have a ActualSink class, maybe it's a good idea to add a method there -> setImmediateOutput. It would look better imo